pandas/tests/io/parser/usecols/test_usecols_basic.py

```diff
@@ -257,13 +257,25 @@ def test_usecols_with_whitespace(all_parsers):
     parser = all_parsers
     data = "a  b  c\n4  apple  bat  5.7\n8  orange  cow  10"
 
+    depr_msg = "The 'delim_whitespace' keyword in pd.read_csv is deprecated"
+
     if parser.engine == "pyarrow":
         msg = "The 'delim_whitespace' option is not supported with the 'pyarrow' engine"
         with pytest.raises(ValueError, match=msg):
-            parser.read_csv(StringIO(data), delim_whitespace=True, usecols=("a", "b"))
+            with tm.assert_produces_warning(
+                FutureWarning, match=depr_msg, check_stacklevel=False
+            ):
+                parser.read_csv(
+                    StringIO(data), delim_whitespace=True, usecols=("a", "b")
```

Comment:
Can most of these be replaced with `sep=r"\s+"` instead already?

Reply:
i think that will introduce a bunch of duplicate tests, this way we can remove these entirely once the deprecation is enforced